net_printer/README.md

```diff
@@ -0,0 +1,116 @@
+file. This program is provided to aid in the development of a contestant router,
+by providing some visibility into the input and output files that contestant
+routers consume and emit.
+
+Consider the following example output:
+
+```
+$ python3 np.py ../vtr_mcml_rwroute.phys u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0
+============================================================
+Route tree for net: u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0
+
+    Source: 0
+    [{      belPin  SLICE_X84Y94 C6LUT O6
+     {      sitePIP SLICE_X84Y94 OUTMUXC D6 False
+            belPin  SLICE_X84Y94 OUTMUXC OUT
+         }  sitePin SLICE_X84Y94 CMUX
+     {   }  belPin  SLICE_X84Y94 OUTMUXC D6
+            sitePin SLICE_X84Y94 C_O
+     {   }  belPin  SLICE_X84Y94 C_O C_O
+            pip     INT_X54Y94 LOGIC_OUTS_W4 INT_NODE_SDQ_49_INT_OUT1 False False
+            pip     INT_X54Y94 INT_NODE_SDQ_49_INT_OUT1 EE2_W_BEG1 False False
+            pip     INT_X55Y94 EE2_W_END1 INT_NODE_SDQ_49_INT_OUT1 False False
+            pip     INT_X55Y94 INT_NODE_SDQ_49_INT_OUT1 SS1_W_BEG1 False False
+            pip     INT_X55Y93 SS1_W_END1 INT_NODE_SDQ_50_INT_OUT1 False False
+            pip     INT_X55Y93 INT_NODE_SDQ_50_INT_OUT1 SS2_W_BEG1 False False
+            pip     INT_X55Y91 SS2_W_END1 INT_NODE_SDQ_54_INT_OUT0 False False
+            pip     INT_X55Y91 INT_NODE_SDQ_54_INT_OUT0 EE4_W_BEG1 False False
+            pip     INT_X57Y91 EE4_W_END1 INT_NODE_SDQ_53_INT_OUT1 False False
+            pip     INT_X57Y91 INT_NODE_SDQ_53_INT_OUT1 EE2_W_BEG2 False False
+            pip     INT_X58Y91 EE2_W_END2 INT_NODE_IMUX_37_INT_OUT0 False False
+            pip     INT_X58Y91 INT_NODE_IMUX_37_INT_OUT0 BYPASS_W8 True False
+            sitePin SLICE_X90Y91 EX
+            belPin  SLICE_X90Y91 EX EX
+     {   }  belPin  SLICE_X90Y91 FFMUXE1 BYP
+            sitePIP SLICE_X90Y91 FFMUXE1 BYP False
+            belPin  SLICE_X90Y91 FFMUXE1 OUT1
+         }] belPin  SLICE_X90Y91 EFF D
+============================================================
+```
+
+In this example `np.py` is invoked on the netlist `vtr_mcml_rwroute.phys` (This
+netlist that can be generated from a global `make`,
+see [the contest website](https://xilinx.github.io/fpga24_routing_contest/index.html))
+in order to printout the net named `u_calc/boundaryChecker/r_ux__57_reg[21]_srl27___u_calc_dropSpin_photon29_o_sleftz_reg_r_n_0`.
+Since this is a fully routed net the output shows one source (`Source: 0`) and
+no stubs. Following the source is a list of all of the edges (FPGAIF
+routeSegements) in the route tree. The symbol `[{` indicates root of the tree.
+The symbol `{` indicates the begining of a branch and the symbol `}` indicates
+the end of a branch. Finally, the symbol `}]` indicates the end of the tree.
+
+In the example above we see that the first edge in the tree is:
```

Comment:
I would suggest being a bit more generic than "edge". It's not clear whether a BELPin is an edge in the routing tree (irrespective of the fact it is captured an edge in the graph you build for analysis) in the same way that a PIP may be more like an edge in the traditional sense. Perhaps "segment" or "element" may be a better fit?

Reply:
how about "edge" -> "routeSegment"?

Reply:
Sure.

Reply:
done: 07921aec937ae653711f00ac66b324c811286cc2